### PR TITLE
feat: Update to be Stacks.js v7 compatible

### PR DESCRIPTION
**Pull request by janniks**
> This PR was published to npm with the alpha versions:
> - connect `npm install @stacks/connect@7.7.2-alpha.cc06270.0 --save-exact`
> - connect-react `npm install @stacks/connect-react@22.4.3-alpha.cc06270.0 --save-exact`
> - connect-ui `npm install @stacks/connect-ui@6.4.2-alpha.cc06270.0 --save-exact`

- update to use both legacy and next network
- update post conditions and clarity serialization
- switch to npm from yarn

### CODE REVIEW

File: packages/connect/README.md [hunk 1]
```diff
@@ -73,12 +73,11 @@ Here, we are sending `10000` micro-STX tokens to a recipient address.
 
 ```js
 import { openSTXTransfer } from '@stacks/connect';
-import { StacksTestnet } from '@stacks/network';
 import { AnchorMode, PostConditionMode } from '@stacks/transactions';
 import { userSession } from './userSession';
 
 openSTXTransfer({
-  network: new StacksTestnet(), // which network to use; use `new StacksMainnet()` for mainnet
+  network: 'testnet', // which network to use; ('mainnet' or 'testnet')
```
Comment (beguene) [inline packages/connect/README.md:80]: Nice, this is much simpler now ! 🎉

**Review by beguene — APPROVED** [review]
Looks great !

If I understand correctly this is backward compatible (with stacks.js v6) at the cost of an increase in bundle size in connect because both versions are imported, which is perfectly fine. But does it mean that stacks v7 is not backward compatible with v6 format ?

Btw I'm curious, why was yarn removed ?

**Comment by janniks** [comment]
Had a weird yarn problem for a while, where we could never recreate the lock file. It was hard to update deps. And npm is the default and everyone has it, one step less 😉

Yep, I think the actual web bundle won't be too much bigger, as it only uses a few imports. But for now we want backwards comp. A next version of Connect would reinvent a bunch of stuff and be non-backwards compatible (while the wallet still are probably), but hopefully more on that front soon!

**Comment by beguene** [comment]
@janniks make sense !

I just wanted to know, as I rarely use yarn nowadays, it's either pnpm or npm. npm is quite fast now, and it's much simpler and predictable.

**Pull request merged**